spec/controllers/build_allocations_controller_spec.rb

```diff
@@ -115,18 +111,13 @@ def further_info
         }
       end
 
-      it 'sends an allocation email with the correct info' do
-        expect(PomMailer).to have_received(:new_allocation_email)
-          .with(
-            offender_no: offender_no,
-            offender_name: "#{offender.fetch(:lastName)}, #{offender.fetch(:firstName)}",
-            pom_name: pom.firstName.capitalize,
-            pom_email: pom.email_address,
-            responsibility: 'responsible',
-            message: notes,
-            url: Rails.application.routes.default_url_options[:host] + prison_prisoner_allocation_path(prison, offender_no),
-            further_info: further_info
-          )
+      it 'sends the correct emails' do
+        expect(EmailService).to have_received(:send_email).with(
+          message: notes,
+          allocation: anything,
+          pom_nomis_id: pom.staffId,
+          further_info: further_info,
+        )
```

Comment:
Does this not reduce specificity/functional coverage of the tests?

Reply:
It's not the job of the BuildAlloationsController unit test to test its dependencies. Thankfully EmailService has its own tests so they already started testing correctly, but then went and turned the controller unit test into a full-stack test.